docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,303 @@
+            "size": 12,
+        }
+        }
+    ]
+}
+```
+
+### oras attach
+
+Attach two files to an image and show the descriptor of the attached files in JSON format.
+
+```bash
+oras attach $REGISTRY/$REPO:$TAG --artifact-type example/sbom sbom.spdx --artifact-type example/vul-scan vul-report.json  --format '{{toPrettyJson .}}'
+```
+
+```json
+{
+    "files" : [
+        {
+            "mediaType": "application/vnd.oci.image.manifest.v1+json",
+            "digest": "sha256:d2a84f4b8b650937ec8f73cd8be2c74add5a911ba64df27458ed8229da804a26",
+            "size": 12,
+            "annotations": {
+                "org.opencontainers.image.created":"2023-11-29T06:32:43Z"
+            },
+            "artifactType" : "example/sbom",
+            "ref": "$REGISTRY/$REPO@$DIGEST_1"
+        },
+        {
+            "mediaType": "application/vnd.oci.image.manifest.v1+json",
+            "digest": "sha256:d2a84f4b8b650937ec8f73cd8be2c74add5a911ba64df27458ed8229da804a27",
+            "size": 12,
+            "annotations": {
+                "org.opencontainers.image.created":"2023-11-29T06:32:43Z"
+            },
+            "artifactType" : "example/vul-scan",
+            "ref": "$REGISTRY/$REPO@$DIGEST_2"
+        }
+    ]
+}
+```
+
+### oras push
+
+Push two files to a repository and show the descriptor of the pushed files in pretty JSON format.
+
+```bash
+oras push $REGISTRY/$REPO:$TAG  --format '{{toPrettyJson .}}'
+```
+
+```json
+{
+    "files" : [
+        {
+            "mediaType": "application/vnd.oci.image.layer.v1.tar",
+            "digest": "sha256:d2a84f4b8b650937ec8f73cd8be2c74add5a911ba64df27458ed8229da804a26",
+            "size": 12,
+            "annotations": {
+                "org.opencontainers.image.title": "hello.txt"
+            },
+            "artifactType": "application/vnd.example+type",
+            "ref": "$REGISTRY/$REPO@$DIGEST"
+        },
+        {
+            "mediaType": "application/vnd.oci.image.layer.v1.tar",
+            "digest": "sha256:d2a84f4b8b650937ec8f73cd8be2c74add5a911ba64df27458ed8229da804a27",
+            "size": 12,
+            "annotations": {
+                "org.opencontainers.image.title": "hello.txt"
+            },
+            "artifactType": "application/vnd.example+type",
+            "ref": "$REGISTRY/$REPO@$DIGEST"
+        }
+    ]
+}
+```
+
+### oras discover
+
+Discover an artifact's referrers. The default output should be listed in a tree view.
+
+```bash
+oras discover localhost:5000/hello:v1
+```
+
+```console
+localhost:5000/hello/demo@sha256:04beb34cd24389147b4642a828b47fabefa722dea794dc3834567cf014ab0fe6
+└── application/vnd.oci.empty.v1+json
+    ├── sha256:1b82e249d83eb4881b8bf4ff9cf13a28799907ddc624b4c3c9140fa77d54fa42
+    ├── sha256:28653e2bb5b5a75393c3a8b58ed9998796299b41dc1ff1f55b9f0844ad7ba39c
+```
+
+Discover an artifact's referrers manifest in pretty JSON. 
+
+```bash
+oras discover localhost:5000/hello:v1 --format '{{toPrettyJson .}}'
+```
+
```

Comment:
Consider removing the `toPrettyJson` function here. I think its ok to document that we can invoke the toPrettyJson but don't recommend it samples and I would just use --format json

Reply:
Removed '{{toPrettyJson .}}' from all samples of this Spec